src/main/java/ch/njol/skript/expressions/ExprScript.java

```diff
@@ -1,93 +1,141 @@
+	"\tbroadcast \"Loaded %the current script%\"",
 	"on script load:",
 	"\tset {running::%script%} to true",
 	"on script unload:",
-	"\tset {running::%script%} to false"
+	"\tset {running::%script%} to false",
+	"set {script} to the script named \"weather.sk\"",
+	"loop the scripts in directory \"quests/\":",
+	"\tenable loop-value"
 })
 @Since("2.0")
-@Events("Script Load/Unload")
-public class ExprScript extends SimpleExpression<String> {
-	
+public class ExprScript extends SimpleExpression<Script> {
+
 	static {
-		Skript.registerExpression(ExprScript.class, String.class, ExpressionType.SIMPLE,
-			"[the] script[['s] name]",
-			"name of [the] script"
+		Skript.registerExpression(ExprScript.class, Script.class, ExpressionType.SIMPLE,
+			"[the] [current] script",
+			"[the] script[s] [named] %strings%",
+			"[the] scripts in [directory|folder] %string%"
 		);
 	}
-	
-	@SuppressWarnings("NotNullFieldNotInitialized")
-	private String name;
-	
+
+	private @Nullable Script script;
+	private @Nullable Expression<String> name;
+	private int mode;
+
 	@Override
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		ParserInstance parser = getParser();
-		if (!parser.isActive()) {
-			Skript.error("You can't use the script expression outside of scripts!");
-			return false;
+		this.mode = matchedPattern;
+		if (matchedPattern == 0) {
+			ParserInstance parser = this.getParser();
+			if (!parser.isActive()) {
+				Skript.error("You can't use the current script expression outside of scripts!");
+				return false;
+			}
+			this.script = parser.getCurrentScript();
+		} else {
+			//noinspection unchecked
+			this.name = (Expression<String>) exprs[0];
 		}
-		String name = parser.getCurrentScript().getConfig().getFileName();
-		if (name.contains("."))
-			name = name.substring(0, name.lastIndexOf('.'));
-		this.name = name;
 		return true;
 	}
-	
+
 	@Override
-	protected String[] get(Event event) {
-		return new String[]{name};
+	protected Script[] get(Event event) {
+		if (script != null)
+			return new Script[]{script};
+		assert name != null;
+		if (mode == 2) {
+			@Nullable String string = name.getSingle(event);
+			if (string == null)
+				return new Script[0];
+			File folder = new File(Skript.getInstance().getScriptsFolder(), string);
+			List<Script> scripts = new ArrayList<>();
+			if (!folder.isDirectory())
+				return new Script[0];
+			this.getScripts(folder, scripts);
+			return scripts.toArray(new Script[0]);
+		}
+		return name.stream(event)
+				.map(SkriptCommand::getScriptFromName)
+				.map(ExprScript::getHandle)
+				.filter(Objects::nonNull)
+				.toArray(Script[]::new);
 	}
-	
+
+	private void getScripts(File folder, List<Script> scripts) {
+		File[] files = folder.listFiles();
+		if (files == null)
+			return;
+		for (File file : files) {
+			if (file.isDirectory()) {
+				this.getScripts(file, scripts);
+			} else if (file.getName().endsWith(".sk")) {
```

Comment:
It might be better to instead check whether it is accepted by either filter in ScriptLoader